src/Stratis.Features.FederatedPeg/TargetChain/FederatedPegBroadcaster.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Stratis.Bitcoin.Connection;
+using Stratis.Bitcoin.P2P.Peer;
+using Stratis.Bitcoin.P2P.Protocol.Payloads;
+using Stratis.Features.FederatedPeg.Interfaces;
+using Stratis.Features.FederatedPeg.NetworkHelpers;
+
+namespace Stratis.Features.FederatedPeg.TargetChain
+{
+    public class FederatedPegBroadcaster : IFederatedPegBroadcaster
+    {
+        private readonly IConnectionManager connectionManager;
+        private readonly IFederatedPegSettings federatedPegSettings;
+
+        public FederatedPegBroadcaster(
+            IConnectionManager connectionManager,
+            IFederatedPegSettings federatedPegSettings)
+        {
+            this.connectionManager = connectionManager;
+            this.federatedPegSettings = federatedPegSettings;
+        }
+
+        /// <inheritdoc />
+        public async Task BroadcastAsync(Payload payload)
+        {
+            // TODO: Optimize how federation peers are identified.
+
+            List<INetworkPeer> peers = this.connectionManager.ConnectedPeers.ToList();
+
+            var ipAddressComparer = new IPAddressComparer();
+
+            foreach (INetworkPeer peer in peers)
+            {
+                // Broadcast to peers.
+                if (!peer.IsConnected)
+                    continue;
+
+                if (this.federatedPegSettings.FederationNodeIpEndPoints.Any(e => ipAddressComparer.Equals(e.Address, peer.PeerEndPoint.Address)))
```

Comment:
@codingupastorm we have `.Match()` extension that we use to match Ip's in `Stratis.Bitcoin.Utilities.Extensions`

Reply:
Can fix in a subsequent PR :)

Reply:
Okay great, I think we want a dictionary or something of IPs too

Reply:
@codingupastorm we might as well change this now seeing as there are more changes required on this PR.